### ISSUE TITLE
Docker Swarm Mode: traefik.docker.network should default only to networks that traefik is attached to

### ISSUE THREAD

**Issue by dfyx** [containous/traefik#979]
Assume the following situation:

* I'm using docker 1.12 in swarm mode
* overlay network `traefik-net` with 192.168.0.0/24
* overlay network `backend` with 192.168.1.0/24
* the traefik container is only attached to the `traefik-net` network with the ip address 192.168.0.2
* my application is attached to both networks because it needs to connect to a database container. Therefore it has two ip addresses: 192.168.0.4 and 192.168.1.2
* the application has the following labels set: `traefik.port=80` and `traefik.frontend.rule=myapp.example.com`

**Expected behavior:**
Traefik creates a backend for http://192.168.0.4

**Observed behavior in 1.1.2:**
Traefik alternates between http://192.168.0.4 and http://192.168.1.2 every time it refreshes its config which sporadically causes 502 Bad Gateway errors. This is clearly visible when refreshing the traefik web interface every 30 seconds or so.

**Possible solutions/fixes:**
* My current workaround: explicitly set `traefik.docker.network=traefik-net` for every single application that is attached to more than one network
* Simple solution: Allow setting a default value for `traefik.docker.network` in `traefik.toml` (as far as I can see from the docs this is not possible right now)
* The "it just works" solution: only consider IP addresses that lie within networks that traefik is attached to

**Comment by dfyx** [comment]
This is related to #959 and a followup to the closed #970

**Comment by michaelkrog** [comment]
I concur. I am exposed to this issue almost every time I create a new service that needs access to multiple networks. 

I simply forget that extra label every time as it works perfectly fine without it for services that only binds to the same network as traefik does.

**Comment by centerorbit** [comment]
I too have encountered this problem, but it exists with local docker-compose setup (no Swarm involved) as well.

I suspect the culprit is in `getIPAddress` of `provider/docker/docker.go`:

```
for _, network := range container.NetworkSettings.Networks {
	return network.Addr
}
```
You'll notice that earlier in the function it attempts to pick a network based on things (such as the label). If no other option exists, it simply loops through all of the Containers possible networks, and picks the first one and returns it.

**Comment by nim65s** [comment]
Hi,
I think this should be closed by https://github.com/containous/traefik/pull/3471

**Comment by mmatur** [comment]
Closed by #3471.

Thanks @nim65s
